drivers/mbox/mbox_nxp_imx_mu.c

```diff
@@ -118,66 +127,66 @@ static const struct mbox_driver_api nxp_imx_mu_driver_api = {
 	.set_enabled = nxp_imx_mu_set_enabled,
 };
 
-#define MU_INSTANCE_DEFINE(idx)							\
-	static struct nxp_imx_mu_data nxp_imx_mu_##idx##_data;			\
-	static struct nxp_imx_mu_config nxp_imx_mu_##idx##_config = {		\
-		.base = (MU_Type *)DT_INST_REG_ADDR(idx),			\
-	};									\
-										\
-	void MU_##idx##_IRQHandler(void);					\
-	static int nxp_imx_mu_##idx##_init(const struct device *dev)		\
-	{									\
-		ARG_UNUSED(dev);						\
-		MU_Init(nxp_imx_mu_##idx##_config.base);			\
-		IRQ_CONNECT(DT_INST_IRQN(idx),					\
-			    DT_INST_IRQ(idx, priority),				\
-			    MU_##idx##_IRQHandler,				\
-			    NULL,						\
-			    0);							\
-		irq_enable(DT_INST_IRQN(idx));					\
-		return 0;							\
-	}									\
-	DEVICE_DT_INST_DEFINE(idx, nxp_imx_mu_##idx##_init, NULL,		\
-			&nxp_imx_mu_##idx##_data, &nxp_imx_mu_##idx##_config,	\
-			POST_KERNEL, CONFIG_MBOX_INIT_PRIORITY,			\
-			&nxp_imx_mu_driver_api)
-
-#define MU_IRQ_HANDLER(idx)							\
-	static uint32_t mu_##idx##_received_data;				\
-	void MU_##idx##_IRQHandler(void)					\
-	{									\
-		const struct device *dev = DEVICE_DT_INST_GET(idx);		\
-		const struct nxp_imx_mu_data *data = dev->data;			\
-		const struct nxp_imx_mu_config *config = dev->config;		\
-		struct mbox_msg msg;						\
-		struct mbox_msg *callback_msg_ptr = NULL;			\
-		uint32_t flag = MU_GetStatusFlags(config->base);		\
-										\
-		for (int i_channel = 0; i_channel < MU_MAX_CHANNELS; i_channel++) { \
-			if ((flag & (kMU_Rx0FullFlag >> i_channel)) == \
-				(kMU_Rx0FullFlag >> i_channel)) { \
-				mu_##idx##_received_data =				\
-					MU_ReceiveMsgNonBlocking(config->base, 0);	\
-				msg.data = (const void *)&mu_##idx##_received_data;	\
-				msg.size = MU_MBOX_SIZE;				\
-				callback_msg_ptr = &msg;				\
-			} else if ((flag & (kMU_GenInt0Flag >> i_channel)) == \
-						(kMU_GenInt0Flag >> i_channel)) {	\
-				MU_ClearStatusFlags(config->base, \
-					(kMU_GenInt0Flag >> i_channel)); \
-				callback_msg_ptr = NULL;				\
-			}								\
-											\
-			if (data->cb[i_channel]) {					\
-				data->cb[i_channel](dev, i_channel,				\
-						data->user_data[i_channel],		\
-						callback_msg_ptr);			\
-			} \
-		} \
+#define MU_INSTANCE_DEFINE(idx)                                                                    \
+	static struct nxp_imx_mu_config nxp_imx_mu_##idx##_config = {                              \
+		.base = (MU_Type *)DT_INST_REG_ADDR(idx),                                          \
+	};                                                                                         \
+	static struct nxp_mu_device_descriptor                                                     \
+		nxp_mu_device_table[DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)] = {0};                 \
```

Comment:
Hi, Thanks for pointing on this. I wanted to only consolidate into one struct, each instance config and data are still stored in structs but are in this wrapper table, instead of being standalone as `static struct nxp_imx_mu_data nxp_imx_mu_##idx##_data;`

And with that in mind I forgot also this to move it there properly:
Instead of 
```c
static struct nxp_imx_mu_config nxp_imx_mu_##idx##_config = {                              \
		.base = (MU_Type *)DT_INST_REG_ADDR(idx),                                          \
	};    
...
nxp_mu_device_table[idx].config = nxp_imx_mu_##idx##_config;
```

Better options would be in just init function (and remove generated static struct nxp_imx_mu_config nxp_imx_mu_##idx##_config):

```c
nxp_mu_device_table[idx].config.base = (MU_Type *)DT_INST_REG_ADDR(idx),                       \
```

I will change this, In case you are OK using this "wrapper" struct, instead of separate struct per instance?

Reply:
I would prefer we do not use a wrapper struct. The advantage of a separate config and data structure is that const elements (like the base address of the IP, in most cases) can be stored in ROM. With the current wrapper struct approach, this data will still be in RAM when it does not need to be.

Reply:
OK That make sense, I have changed it as recommenced :)